lib/cyclonedx/cocoapods/pod.rb

```diff
@@ -0,0 +1,75 @@
+require 'rubygems/version'
+require_relative 'license'
+
+module CycloneDX
+  module CocoaPods
+    class Pod
+      attr_reader :name        # xs:normalizedString
+      attr_reader :version     # xs:normalizedString
+      attr_reader :checksum    # https://cyclonedx.org/docs/1.2/#type_hashValue (We only use SHA-1 hashes - length == 40)
+      attr_reader :author      # xs:normalizedString
+      attr_reader :description # xs:normalizedString
+      attr_reader :license     # https://cyclonedx.org/docs/1.2/#type_licenseType
+                               # We don't currently support several licenses or license expressions https://spdx.github.io/spdx-spec/appendix-IV-SPDX-license-expressions/
+      def initialize(name:, version:, checksum: nil)
+        raise ArgumentError, "Name must be non empty" if name.nil? || name.to_s.strip.empty?
+        Gem::Version.new(version) # To check that the version string is well formed
+        raise ArgumentError, "#{checksum} is not valid SHA-1 hash" unless checksum.nil? || checksum =~ /[a-fA-F0-9]{40}/
+        @name, @version, @checksum = name.to_s.strip, version, checksum
+      end
+
+      def populate(attributes)
+        attributes.transform_keys!(&:to_sym)
+        populate_author(attributes)
+        populate_description(attributes)
+        populate_license(attributes)
+      end
+
+      def purl
+        "pkg:pod/#{name}@#{version}"
```

Comment:
This has yet to be defined. https://github.com/package-url/purl-spec. My guess is that it will be 'cocoapods'.

Reply:
Are names case sensitive? Are there any special characters that may need to be escaped?

Reply:
Ok, I'll change it to cocoapods.

Regarding names, I thinks they are case sensitive, and they are usually CamelCase but can contain dots, dashes, underscores... should this be escaped? Would `CGI.escape` be enough for this?

Reply:
Names URL encoded (just in case) in 03de233afbe4e0c35719b7502be3b6f5186a5992
`cocoapods` instead of `pod` in d71aef4f2e09a7e5c650ed75dbd4b3b9561e8226